samples/scripts/PowerShell/GenerateOpenApiYaml.ps1

```diff
@@ -0,0 +1,33 @@
+<#
+.SYNOPSIS 
+Generates the OpenApi doc for the specified version and compares it with the baseline to make sure no breaking changes are introduced
+.Parameter WorkingDir
+The working directory
+.PARAMETER OutputPathofDll
+Path for the web projects dll
+.PARAMETER OutputPathOfOpenApiDoc
+Path to output the OpenApi Doc (in yaml)
+.PARAMETER Versions
+Api versions to generate the OpenApiDoc for and compare with baseline
+#>
+
+param(
+    [string]$WorkingDir,
+
+    [string]$OutputPathofDll,
+
+    [string]$OutputPathOfOpenApiDoc,
+
+    [String[]] $Versions
+)
+foreach($Version in $Versions){
+    write-host "Generating Yaml file for $Version"
+
+    dotnet new tool-manifest --force
+    dotnet tool install --version 6.1.4 Swashbuckle.AspNetCore.Cli
+    dotnet swagger tofile --yaml --output $WorkingDir/$OutputPathOfOpenApiDoc/$Version.yaml $WorkingDir/$OutputPathofDll $Version
+
+    write-host "Running comparison with baseline for version $Version"
+    docker run --rm -t -v ${WorkingDir}:/dicom-server openapitools/openapi-diff:latest /dicom-server/swagger/$Version/swagger.yaml /dicom-server/$OutputPathOfOpenApiDoc/$version.yaml --fail-on-incompatible
```

Comment:
So for versions that are still in development / unreleased, /dicom-server/swagger/$Version/swagger.yaml can be changed whenever. As soon as the version gets released, that file shouldn't be changed at all so this comparison diff will work. Is that right?

Reply:
If so I wonder if we can lock the file somehow after release, so it isn't accidentally updated

Reply:
Yes that is correct. I am not sure if there is anyway to lock it, but I was thinking that is something that the people reviewing the pr should notice? Or do you have any ideas on some kind of automatic check

Reply:
I think we had chatted about this, and I think there are probably two ways to perform this check:
1. Generate each version's OpenAPI YAML specification dynamically from the assembly and compare that against a versioned control. If it's at all changed, I'd say we fail the build because it is out-of-sync, and instead force the developer to re-generate it through some script such that the change is reviewed in PR.
2. We never generate the YAML manually. Instead, we always auto-generate it as part of the build and we compare the auto-generated files from the current build to the auto-generated files from the previous build. We can fail the build if there is a breaking change, while using monikers like `prerelease` to determine when they're allowed.

Reply:
As discussed, we are following approach 1 currently